src/components/header/Header.tsx

```diff
@@ -354,21 +396,113 @@ const Header = () => {
                 borderRadius: "10px",
                 outline: "none",
                 fieldset: { border: "none", outline: "none" },
+               
               }}
             >
               <SearchIcon sx={{ ml: { xl: "10px" } }} />
-              <TextField
-                placeholder="Find your perfect deal"
+           
+             
+              <Autocomplete
+              className="xyz"
+              value={searchValue}
+              open={openListBox}
+              disableClearable
+              onFocus={()=>{
+                if(searchValue){
+                  setOpenLIstBox(true)
+                }
+              }}
+
+              PopperComponent={(props) => (
+                <Popper {...props} style={{ width: "28.75rem" }}>
+                  {props.children}
+                </Popper>
+              )}
+              onClose={()=>setOpenLIstBox(false)}
+               
+                freeSolo
+                id="country-select-demo"
                 sx={{
-                  height: "100%",
-                  width: "25.4rem",
-                  border: "none",
-                  // bgcolor:"blue",
-                  justifyContent: "center",
-                  fieldset: { border: "none", outline: "none" },
+                  width: "24.7rem",
+                  "MuiAutocomplete-loading": { display: "none" }, "&:ul":{background:"pink", width:"300%"}
                 }}
-              ></TextField>
-              <CloseIcon sx={{ mr: { xl: "0.5rem" } }} />
+                filterOptions={(options) => options}
+                options={
+                  searchResultData.length > 0
+                    ? searchResultData
+                    : [
+                        
+                      ]
+                }
+             renderOption={(props, searchResultData) => (
+               searchValue && <Box
+                 component="li"
+                 sx={{
+                   display: "flex",
+                   flexDirection: "row",
+                   width: "100%", margin:"0px",padding:"0px",
+                   justifyContent: "space-between !important",
+                 }}
+                 {...props}
+               >
+                 <Box
+                   component={"div"}
+                   sx={{ height: "3rem", width: "15%" }}
+                 >
+                   <Box
+                     component={"img"}
+                     src={searchResultData.productImages && searchResultData?.productImages?.[0]?.imageUrl}
```

Comment:
don't need to check `searchResultData.productImages` because you have already added `?`